eglot-jl.jl

```diff
@@ -1,13 +1,33 @@
-# Make sure that we only load packages from this environment specifically.
+# Usage:
+#   julia --project path/to/eglot-jl/eglot-jl.jl [DEPOT_PATH]
+
+# Get the project environment.
+#
+# WARNING: this script must be called with the `--project` command-line switch
+# for this to work reliably.
+project_path = dirname(Base.load_path()[1])
```

Comment:
Ah. I think you were confused. There are two projects we care about here:
- the project in which the language server runs (needs to be the Project.toml included with this emacs package)
- the target project the language server is running against

We can't run the language server in the target environment.

Reply:
Nevermind. I see that below you've added an explicit `Pkg.activate(@__DIR__)`. So behavior should be correct. And `Pkg` should be available since importing `Pkg` comes before we manipulate the `LOAD_PATH`. And I see how it's convenient to just use the `--project` flag for this so that it lets julia figure out the directory itself. It just feels very hacky; let me sleep on whether I think this is a good idea or not.

Reply:
I do understand that it feels hacky, and I was reluctant to do this myself at first: I was going to first get the project path by running something along the lines of:
`julia --project -e 'println(dirname(Base.load_path()[1]))`

Then I would resume the current way of doing things and call
`julia --project=/path/to/eglot-jl /path/to/eglot-jl/eglot-jl.jl /path/to/user/project`

The way I convinced myself to do it like this is the following: non-power-users of julia always start a REPL with a simple `julia` (without any `--project` argument), and call `Pkg.activate(...)` first thing in the session. The same can be said of users of IDEs such as Juno, who let Juno start the REPL for them and have to activate their project themselves (unless they do some form of non-standard configuration of the IDE).

So I thought it actually was kind of OK to `Pkg.activate()` the correct project after having started Julia. And why not display the path to the user project first?

But anyways, I'm still not so sure about this myself. If you find this way too hacky, we can always revert to the first proposal: call julia once before in order to get the user project path, and leave everything else as it currently is.